### PR TITLE
Swap

### PR DESCRIPTION
**Pull request by juan-cortes**
image

Everything should be ready for testing now. 
Everything should be working now.
If you break something, please add a subtask at https://ledgerhq.atlassian.net/browse/LL-3129

### CODE REVIEW

File: src/renderer/screens/swap/Form/To.js [hunk 1]
```diff
@@ -0,0 +1,235 @@
...
+  } = useCurrencyAccountSelect({
+    allCurrencies: currencies,
+    allAccounts: validAccounts,
+    defaultCurrency,
+    defaultAccount,
+    fallbackToFirst: false,
+  });
+
+  const unit = currency && currency.units[0];
+  const renderOptionOverride = ({ data: currency }: any) => {
```
Comment (valpinkman) [inline src/renderer/screens/swap/Form/To.js:96]: maybe `useCallback` here ?

File: src/renderer/screens/swap/History/OperationRow.js [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+  const fromCurrency = getAccountCurrency(fromAccount);
+  const toCurrency = getAccountCurrency(toAccount);
+
+  return (
+    <Row
+      toExists={toExists}
+      horizontal
+      key={swapId}
+      alignItems={"center"}
+      onClick={() => openSwapOperationDetailsModal(mappedSwapOperation)}
```
Comment (valpinkman) [inline src/renderer/screens/swap/History/OperationRow.js:120]: useCallback

**Review by valpinkman — DISMISSED** [review]

File: src/renderer/components/PerCurrencySelectAccount/state.js [hunk 3]
```diff
@@ -48,14 +48,19 @@ export function useCurrencyAccountSelect({
   allAccounts,
   defaultCurrency,
   defaultAccount,
+  fallbackToFirst = true,
```
[the remark this answers is not in the document]
Comment (juan-cortes) [reply src/renderer/components/PerCurrencySelectAccount/state.js:51]: Probably nowhere now, it was initially used and then product wanted to not preselect anything. good catch

[the remark this answers is not in the document]
Comment (juan-cortes) [reply src/renderer/components/PerCurrencySelectAccount/state.js]: Solved on latest

File: src/renderer/screens/swap/Form/To.js (hunk 1 above)
Comment (juan-cortes) [reply src/renderer/screens/swap/Form/To.js]: You were actually crucial in finding a re render loop bug thingy, so thank you

**Pull request merged**